homeassistant/components/hyperion/camera.py

```diff
@@ -0,0 +1,255 @@
+        """Add entities for a new Hyperion instance."""
+        assert server_id
+        async_add_entities(
+            [
+                HyperionCamera(
+                    server_id,
+                    instance_num,
+                    instance_name,
+                    entry_data[CONF_INSTANCE_CLIENTS][instance_num],
+                )
+            ]
+        )
+
+    @callback
+    def instance_remove(instance_num: int) -> None:
+        """Remove entities for an old Hyperion instance."""
+        assert server_id
+        async_dispatcher_send(
+            hass,
+            SIGNAL_ENTITY_REMOVE.format(
+                camera_unique_id(instance_num),
+            ),
+        )
+
+    listen_for_instance_updates(hass, config_entry, instance_add, instance_remove)
+    return True
+
+
+# A note on Hyperion streaming semantics:
+#
+# Different Hyperion priorities behave different with regards to streaming. Colors will
+# not stream (as there is nothing to stream). External grabbers (e.g. USB Capture) will
+# stream what is being captured. Some effects (based on GIFs) will stream, others will
+# not. In cases when streaming is not supported from a selected priority, there is no
+# notification beyond the failure of new frames to arrive.
+
+
+class HyperionCamera(Camera):
+    """ComponentBinarySwitch switch class."""
+
+    def __init__(
+        self,
+        server_id: str,
+        instance_num: int,
+        instance_name: str,
+        hyperion_client: client.HyperionClient,
+    ) -> None:
+        """Initialize the switch."""
+        super().__init__()
+
+        self._unique_id = get_hyperion_unique_id(
+            server_id, instance_num, TYPE_HYPERION_CAMERA
+        )
+        self._name = f"{instance_name} {NAME_SUFFIX_HYPERION_CAMERA}".strip()
+        self._device_id = get_hyperion_device_id(server_id, instance_num)
+        self._instance_name = instance_name
+        self._client = hyperion_client
+
+        self._image_cond = asyncio.Condition()
+        self._image: bytes | None = None
+
+        # The number of open streams, when zero the stream is stopped.
+        self._image_stream_clients = 0
+
+        self._client_callbacks = {
+            f"{KEY_LEDCOLORS}-{KEY_IMAGE_STREAM}-{KEY_UPDATE}": self._update_imagestream
+        }
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique id for this instance."""
+        return self._unique_id
+
+    @property
+    def name(self) -> str:
+        """Return the name of the switch."""
+        return self._name
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if the camera is on."""
+        return self.available
+
+    @property
+    def available(self) -> bool:
+        """Return server availability."""
+        return bool(self._client.has_loaded_state)
+
+    @callback
+    async def _update_imagestream(self, img: dict[str, Any] | None = None) -> None:
```

Comment:
async methods cannot be callbacks.

```suggestion
    async def _update_imagestream(self, img: dict[str, Any] | None = None) -> None:
```